kernel/src/platform/scheduler_timer.rs

```diff
@@ -117,8 +109,6 @@ impl SchedulerTimer for () {
 
     fn disarm(&self) {}
 
-    fn arm(&self) {}
-
```

Comment:
Given that disarm is now not called from `sched.rs`, does it still make sense for it to be part of this trait?

Reply:
If you think so, sure, although perhaps that should be a separate PR on main which we can merge into this?

Reply:
this PR is the one that removes the use of `arm()` and `disarm()` from `sched.rs`, I can break all those changes out into a separate PR if you like, or just followup with that change once this merges.